linera-service/src/linera.rs

```diff
@@ -1835,6 +1841,60 @@ impl Runnable for Job {
                 _ => unreachable!("other project commands do not require storage"),
             },
 
+            Wallet(WalletCommand::Init {
+                faucet: Some(faucet_url),
+                ..
+            }) => {
+                info!("Requesting a new chain from the faucet.");
+                let key_pair = context.generate_key_pair();
+                let public_key = key_pair.public();
+                context.wallet_state.add_unassigned_key_pair(key_pair);
+                let outcome = cli_wrappers::Faucet::claim_url(&public_key, &faucet_url).await?;
+                println!("{}", outcome.chain_id);
+                println!("{}", outcome.message_id);
+                println!("{}", outcome.certificate_hash);
+
+                let state = WorkerState::new("Local node".to_string(), None, storage)
+                    .with_allow_inactive_chains(true)
+                    .with_allow_messages_from_deprecated_epochs(true);
+                let mut node_client = LocalNodeClient::new(state, Notifier::default());
+
+                let admin_chain_id = context.wallet_state.genesis_admin_chain();
+                let query = ChainInfoQuery::new(admin_chain_id).with_committees();
+                let info = node_client.handle_chain_info_query(query).await?;
+                let committee = info
+                    .latest_committee()
+                    .context("Invalid chain info response; missing latest committee")?;
+                let nodes = context.make_node_provider().make_nodes(committee)?;
+
+                // Download the parent chain.
+                let target_height = outcome.message_id.height.try_add_one()?;
+                let parent_chain_id = outcome.message_id.chain_id;
+                node_client
+                    .download_certificates(nodes, parent_chain_id, target_height)
+                    .await
+                    .context("failed to download parent chain")?;
+
+                // The initial timestamp for the new chain is taken from the block with the message.
+                let certificate = node_client
+                    .certificate_for(&outcome.message_id)
+                    .await
+                    .context("could not find OpenChain message")?;
+                let timestamp = match certificate.value() {
+                    CertificateValue::ConfirmedBlock {
+                        executed_block: ExecutedBlock { block, .. },
+                        ..
+                    } => block.timestamp,
+                    _ => bail!("Unexpected certificate."),
```

Comment:
Is there any chance we can make this error more precise - so that if it does happen it can help a user understand how to start to debug this? What action can a user take to amend this? Why might this happen etc.

There are a bunch of errors in `linera` which are not very "useful" and I would like to improve this.

Reply:
They are precise but do not help someone who is not an expert in the internals of the system understand what is happening.

Reply:
I added a bit more information, but in this case, what _can_ you do? It's probably either a bug, or some surprising version incompatibility, or a defective or fake network.